### PR TITLE
fix(datastore): Fix subscription creation with model subscription level is public

### PR DESCRIPTION
**Pull request by onlybakam**
_Issue #, if available:_ https://github.com/aws-amplify/amplify-js/issues/5173

_Description of changes:_

Create subscriptions with no "owner" auth check when the model subscription level is public and an owner auth rule is used. This aligns the datastore behavior with the subscription definition created by the Amplify cli.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mdoesburg** [comment]
@manueliglesias Can you review this please?

File: packages/datastore/src/sync/utils.ts [hunk 1]
```diff
@@ -183,15 +183,30 @@ export function getAuthorizationRules(
...
 				groups,
+				areSubscriptionsPublic: false,
 			};
 
+			if (authStrategy === 'owner') {
+				// look for the subscription level override
+				// only pay attention to the public level
+				const modelConfig = []
+					.concat(modelDefinition.attributes)
+					.find(attr => attr && attr.type === 'model');
```
Comment (manueliglesias) [inline packages/datastore/src/sync/utils.ts:201]:
```suggestion
				const modelConfig = (<typeof modelDefinition.attributes>[])
					.concat(modelDefinition.attributes)
					.find(attr => attr && attr.type === 'model');
```

File: packages/datastore/src/sync/utils.ts [hunk 2]
```diff
@@ -183,15 +183,30 @@ export function getAuthorizationRules(
...
+
+				// find the subscriptions level. ON is defeault
```
Comment (manueliglesias) [inline packages/datastore/src/sync/utils.ts:203]:
```suggestion
				// find the subscriptions level. ON is default
```

File: packages/datastore/__tests__/subscription.test.ts [hunk 3]
```diff
@@ -85,6 +85,91 @@ describe('sync engine subscription module', () => {
 			)
 		).toEqual(authInfo);
 	});
+	test('owner authorization with public subscription', () => {
+		const model = {
```
Comment (manueliglesias) [inline packages/datastore/__tests__/subscription.test.ts:89]:
```suggestion
		const model: SchemaModel = {
```

File: packages/datastore/__tests__/subscription.test.ts [hunk 4]
```diff
@@ -85,6 +85,91 @@ describe('sync engine subscription module', () => {
...
+		};
+		const authInfo = {
+			authMode: 'AMAZON_COGNITO_USER_POOLS',
+			isOwner: false,
+			ownerField: 'owner',
+			ownerValue: 'user1',
+		};
+
+		expect(
+			SubscriptionProcessor.prototype.getAuthorizationInfo(
```
Comment (manueliglesias) [inline packages/datastore/__tests__/subscription.test.ts:165]:
```suggestion
			// @ts-ignore
			SubscriptionProcessor.prototype.getAuthorizationInfo(
```

**Pull request merged**